### PR TITLE
refactor: Landscape membership list

### PR DESCRIPTION
**Pull request by josebui**
## Description

Landscape is using a group to handle the user memberships in this PR I changed it to use MembershipList to removing the dependency with Group.

With this PR there is some duplciation added for the components and functions that eventually will be removed when groups are changed to use membership list too.

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added

### Related Issues
- https://github.com/techmatters/terraso-product/issues/415

### Related PRs
- https://github.com/techmatters/terraso-backend/pull/946
- https://github.com/techmatters/terraso-client-shared/pull/153

### CODE REVIEW

File: src/collaboration/components/MemberJoin.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/components/MemberJoin.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/collaborationContext.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/collaborationContext.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/components/MembershipCard.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipCard.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/components/MembershipCard.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  const membersSample = _.getOr([], 'membershipsSample', membershipsInfo);
+  const totalCount = _.getOr(0, 'totalCount', membershipsInfo);
+
+  if (fetching) {
+    return <Loader />;
+  }
+
+  const pendingRequest =
+    accountMembership &&
+    accountMembership.membershipStatus === MEMBERSHIP_STATUS_PENDING;
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipCard.js:71]:
```suggestion
  const pendingRequest =
    accountMembership?.membershipStatus === MEMBERSHIP_STATUS_PENDING;
```

File: src/collaboration/components/MembershipCount.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipCount.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/components/MembershipJoinLeaveButton.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipJoinLeaveButton.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/components/MembershipJoinLeaveButton.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  }, [onMemberJoin, owner]);
+
+  const is_member = useMemo(
+    () => accountMembership?.membershipId,
+    [accountMembership]
+  );
+
+  if (
+    is_member &&
+    accountMembership.membershipStatus === MEMBERSHIP_STATUS_PENDING
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipJoinLeaveButton.js:54]:
```suggestion
    accountMembership?.membershipStatus === MEMBERSHIP_STATUS_PENDING
```

File: src/collaboration/components/MembershipPendingWarning.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/collaboration/components/MembershipPendingWarning.js:2]:
```suggestion
 * Copyright © 2023 Technology Matters
```

File: src/collaboration/components/RoleSelect.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import React, { useCallback } from 'react';
```
Comment (paulschreiber) [inline src/collaboration/components/RoleSelect.js:1]: issue: add copyright notice.

File: src/landscape/landscapeService.js [hunk 10]
```diff
@@ -15,22 +15,20 @@
...
-import { countryNameForCode } from 'common/countries';
-import * as gisService from 'gis/gisService';
-import { extractDataEntries } from 'sharedData/sharedDataUtils';
 import { extractTerms } from 'taxonomies/taxonomiesUtils';
 
-import { ALL_PARTNERSHIP_STATUS } from './landscapeConstants';
+import {
+  ALL_PARTNERSHIP_STATUS,
+  MEMBERSHIP_ROLE_MEMBER,
+} from './landscapeConstants';
```
Comment (paulschreiber) [inline src/landscape/landscapeService.js:27]: issue: do not use relative path.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/landscape/landscapeSlice.js [hunk 11]
```diff
@@ -393,9 +500,32 @@ export const {
...
+const updateMemberItem = (state, email, valueGenerator) => {
+  return _.set(
+    'members.data.membershipsInfo.membershipsSample',
+    state.members.data.membershipsInfo.membershipsSample
+      .map(membership => {
+        if (membership.email === email) {
+          return valueGenerator(membership);
+        }
+        return membership;
+      })
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/landscape/landscapeSlice.js:527]: @paulschreiber IMO your suggestion is better

**Comment by josebui** [comment]
The final PR is: https://github.com/techmatters/terraso-web-client/pull/1317

**Pull request closed without merging**